src/modules/flow/json/json.c

```diff
@@ -1008,11 +1029,318 @@ json_serialize(struct sol_buffer *buffer, struct json_element *element)
     return -EINVAL;
 }
 
+static struct json_key_element *
+json_object_get_or_create_child_element(struct json_element *element, const struct sol_str_slice key_slice)
+{
+    uint16_t i;
+    struct json_key_element *key_element;
+
+    SOL_VECTOR_FOREACH_IDX (&element->children, key_element, i)
+        if (sol_str_slice_str_eq(key_slice, key_element->key))
+            return key_element;
+
+    key_element = sol_vector_append(&element->children);
+    SOL_NULL_CHECK(key_element, NULL);
+    key_element->key = sol_str_slice_to_string(key_slice);
+    SOL_NULL_CHECK_GOTO(key_element->key, str_error);
+
+    key_element->element.type = JSON_TYPE_UNKNOWN;
+
+    return key_element;
+
+str_error:
+    sol_vector_del(&element->children, element->children.len - 1);
```

Comment:
This kind of error handling where the last item is removed happens a lot in with the vector stuff. Might be a nice idea, for another patch, to have a `sol_vector_del_element()` that takes a pointer to an element, calculates the offset, and finally calls `sol_vector_del()`.